### ISSUE TITLE
Produce file processing log

### ISSUE THREAD

**Issue by jappeace** [jappeace/cut-the-crap#16]
At one point I got a video of 20 minute as output from a 4 hour source file, that's obviously wrong. I don't know what happened because I don't have any log files. I probably always should generate a log per file processed at least.

**Comment by badly-drawn-wizards** [comment]
Could you elaborate a bit:
 - What sort of format you expect the logs to be in? What information should be included?
 - Do these logs come from ffmpeg, or the program itself?

**Comment by jappeace** [comment]
yeah capture logs from ffmpeg, but also like which step we're on because we're using various ffmpeg commands. It's important to know which one is failing, if we fail.

So the goal is:
+ setup some kind of logging mechanism
+ record which step we're on and capture output of the step.
+ probably in the process you can cleanup the monad stack (it's hardly a stack right now, and if you know how to)

All at your discretion. Any improvement is welcome.
